core/finality-grandpa/src/communication/gossip.rs

```diff
@@ -512,6 +512,40 @@ enum PendingCatchUp {
 	},
 }
 
+/// Configuration for the round catch-up mechanism.
+enum CatchUpConfig {
+	/// Catch requests are enabled, our node will issue them whenever it sees a
+	/// neighbor packet for a round further than `CATCH_UP_THRESHOLD`. If
+	/// `only_from_authorities` is set, the node will only send catch-up
+	/// requests to other authorities it is connected to. This is useful if the
+	/// GRANDPA observer protocol is live on the network, in which case full
+	/// nodes (non-authorities) don't have the necessary round data to answer
+	/// catch-up requests.
+	Enabled { only_from_authorities: bool },
```

Comment:
In the *sentry* node case, one would set `only_from_authorities` to `false` for validators only publicly routable through *sentry* nodes?

Reply:
This is more related to when the observer protocol is enabled (in which case we need to make the distinction). After we introduce `--sentry`, the nodes will announce themselves as authorities threfore the private/validator nodes will be able to request from the sentries (even when the observer protocol is enabled and this is therefore set to `true`).